src/figuresClasses.ts

```diff
@@ -1,19 +1,81 @@
-export interface Figure {
 
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle'
+}
+
+type Color = 'red' | 'green' | 'blue';
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
 export class Triangle implements Figure {
+  shape = Shape.triangle;
+
+  constructor(
+    public color: Color,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0 || a + b <= c || b + c <= a || c + a <= b) {
+      throw new Error('side is not correct');
+    }
+  }
 
+  getArea(): number {
+    const { a, b, c } = this;
+    const p = (a + b + c) * (1 / 2);
```

Comment:
Rename 'p' because, it's not understandable what is 'p'. Better to call it semiperimitr or halfperemitr. I had the same mistake.